rsyslog/8.2106.0/rsyslog-8.2106.0.patch

```diff
@@ -0,0 +1,455 @@
++	AC_SUBST(WOLFSSL_LIBS,    ["-L${withval}/lib -lwolfssl"])
++	enable_wolfssl=yes
++	AC_DEFINE([ENABLE_OPENSSL], [1], [Indicator that openssl is present])
++	AC_DEFINE([ENABLE_WOLFSSL], [1], [Indicator that wolfssl is present])
++],
++[
++	enable_wolfssl=no
++])
++AM_CONDITIONAL(ENABLE_WOLFSSL, test x$enable_wolfssl = xyes)
+ 
+ # openssl support
+ AC_ARG_ENABLE(openssl,
+@@ -1065,8 +1080,7 @@ if test "x$enable_openssl" = "xyes"; then
+ 	AC_DEFINE([ENABLE_OPENSSL], [1], [Indicator that openssl is present])
+ 	save_libs=$LIBS
+ fi
+-AM_CONDITIONAL(ENABLE_OPENSSL, test x$enable_openssl = xyes)
+-
++AM_CONDITIONAL(ENABLE_OPENSSL, test x$enable_openssl = xyes || test x$enable_wolfssl = xyes)
+ 
+ # GnuTLS support
+ AC_ARG_ENABLE(gnutls,
+@@ -2813,6 +2827,7 @@ echo "    hiredis support enabled:                  $enable_omhiredis"
+ echo
+ echo "---{ protocol support }---"
+ echo "    openssl network stream driver enabled:    $enable_openssl"
++echo "    wolfssl network stream driver enabled:    $enable_wolfssl"
+ echo "    GnuTLS network stream driver enabled:     $enable_gnutls"
+ echo "    GSSAPI Kerberos 5 support enabled:        $enable_gssapi_krb5"
+ echo "    RELP support enabled:                     $enable_relp"
+diff --git a/runtime/Makefile.am b/runtime/Makefile.am
+index f62773952..867d31dca 100644
+--- a/runtime/Makefile.am
++++ b/runtime/Makefile.am
+@@ -207,10 +207,16 @@ endif # if ENABLE_INET
+ if ENABLE_OPENSSL
+ pkglib_LTLIBRARIES += lmnsd_ossl.la
+ lmnsd_ossl_la_SOURCES = nsd_ossl.c nsd_ossl.h nsdsel_ossl.c  nsdsel_ossl.h
+-lmnsd_ossl_la_CPPFLAGS = $(PTHREADS_CFLAGS) $(RSRT_CFLAGS) $(OPENSSL_CFLAGS)
+ lmnsd_ossl_la_LDFLAGS = -module -avoid-version
++
++if ENABLE_WOLFSSL
++lmnsd_ossl_la_CPPFLAGS = $(PTHREADS_CFLAGS) $(RSRT_CFLAGS) $(WOLFSSL_CFLAGS)
++lmnsd_ossl_la_LIBADD = $(WOLFSSL_LIBS)
++else
++lmnsd_ossl_la_CPPFLAGS = $(PTHREADS_CFLAGS) $(RSRT_CFLAGS) $(OPENSSL_CFLAGS)
+ lmnsd_ossl_la_LIBADD = $(OPENSSL_LIBS)
+ endif
++endif
+ 
+ #
+ # GnuTLS netstream driver
+diff --git a/runtime/nsd_ossl.c b/runtime/nsd_ossl.c
+index 79347916e..d9543b6c8 100644
+--- a/runtime/nsd_ossl.c
++++ b/runtime/nsd_ossl.c
+@@ -28,6 +28,9 @@
+ #include <stdlib.h>
+ #include <assert.h>
+ #include <string.h>
++#ifdef ENABLE_WOLFSSL
++#include <wolfssl/options.h>
++#endif
+ #include <openssl/ssl.h>
+ #include <openssl/x509v3.h>
+ #include <openssl/err.h>
+@@ -66,6 +69,9 @@ DEFobjCurrIf(datetime)
+ DEFobjCurrIf(nsd_ptcp)
+ 
+ /* OpenSSL API differences */
++#ifdef ENABLE_WOLFSSL
++#define RSYSLOG_X509_NAME_oneline(X509CERT) X509_get_subject_name(X509CERT)
++#else
+ #if OPENSSL_VERSION_NUMBER >= 0x10100000L
+ 	#define RSYSLOG_X509_NAME_oneline(X509CERT) X509_get_subject_name(X509CERT)
+ 	#define RSYSLOG_BIO_method_name(SSLBIO) BIO_method_name(SSLBIO)
+@@ -77,6 +83,7 @@ DEFobjCurrIf(nsd_ptcp)
+ 	#define RSYSLOG_BIO_number_read(SSLBIO) SSLBIO->num
+ 	#define RSYSLOG_BIO_number_written(SSLBIO) SSLBIO->num
+ #endif
++#endif /* ENABLE_WOLFSSL */
```

Comment:
Should be one line up.

Reply:
The `/* ENABLE_WOLFSSL */`? I think it's fine as-is because `#ifdef ENABLE_WOLFSSL` is the outer guard.

Reply:
`ENABLE_OPENSSL` is the outer guard. `ENABLE_WOLFSSL` is inside of this `if endif` block.

Reply:
I think I got the files mixed up. Is this for `tests/tcpflood.c` or `runtime/nsd_ossl.c`?

Reply:
Nevermind. It's good.